src/swsssdk/dbconnector.py

```diff
@@ -9,182 +9,324 @@
     SONIC_DB_CONFIG_FILE = "/var/run/redis/sonic-db/database_config.json"
+    SONIC_DB_GLOBAL_CONFIG_FILE = "/var/run/redis/sonic-db/database_global.json"
     _sonic_db_config_init = False
     _sonic_db_config = {}
 
     @staticmethod
-    def load_sonic_db_config(sonic_db_file_path=SONIC_DB_CONFIG_FILE):
+    def load_sonic_db_config(sonic_db_file_path=SONIC_DB_GLOBAL_CONFIG_FILE):
         """
-        Get multiple database config from the database_config.json
+        Parse and load the database config json files
+        Check if either of SONIC_DB_GLOBAL_CONFIG_FILE or SONIC_DB_CONFIG_FILE is present, unless the 
+        'sonic_db_file_path' is explicitly passed as parameter.
         """
         if SonicDBConfig._sonic_db_config_init == True:
             return
 
         try:
-            if os.path.isfile(sonic_db_file_path) == False:
-                msg = "'{}' is not found, it is not expected in production devices!!".format(sonic_db_file_path)
-                logger.warning(msg)
-                sonic_db_file_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'config', 'database_config.json')
-            with open(sonic_db_file_path, "r") as read_file:
-                SonicDBConfig._sonic_db_config = json.load(read_file)
+            if os.path.isfile(sonic_db_file_path) ==  True:
+                with open(sonic_db_file_path, "r") as read_file:
+                    all_dbs = json.load(read_file)
+                    for entry in all_dbs['INCLUDES']:
+                        if 'namespace' not in entry.keys():
+                            namespace = 'default'
+                        else: 
+                            namespace = entry['namespace']
+
+                        db_include_file = SonicDBConfig.SONIC_DB_BASE_DIR+entry['include']
+                        SonicDBConfig._sonic_db_config[namespace] = None
+
+                        # Not finding the database_config.json file for the namespace
+                        if os.path.isfile(db_include_file) == False:
+                            msg = "'{}' file is not found !!".format(db_include_file)
+                            logger.warning(msg)
+                            continue
+
+                        with open(db_include_file, "r") as inc_file:
+                            SonicDBConfig._sonic_db_config[namespace] = json.load(inc_file)
+            else:
+                sonic_db_file_path = SonicDBConfig.SONIC_DB_CONFIG_FILE
+                if os.path.isfile(sonic_db_file_path) ==  False:
+                    msg = "'{}' is not found, it is not expected in production devices!!".format(sonic_db_file_path)
+                    logger.warning(msg)
+                    sonic_db_file_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'config', 'database_config.json')
+
+                with open(sonic_db_file_path, "r") as read_file:
+                    SonicDBConfig._sonic_db_config['default'] = json.load(read_file)
```

Comment:
>'default' [](start = 51, length = 9)

`default` is a valid namespace name. let's use `''` as your default.

Reply:
Changed to using '' instead of 'default'